### DISCUSSION TITLE
How to fix "java.util.ServiceConfigurationError: org.junit.platform.engine.TestEngine: org.junit.jupiter.engine.JupiterTestEngine not a subtype"?

### DISCUSSION THREAD

**Discussion by carstenartur** [eclipse-tycho/tycho#1899] [Q&A]
To reproduce issue run "mvn verify" on 

https://github.com/carstenartur/sandbox

I have seen similar errors at
https://bugs.eclipse.org/bugs/show_bug.cgi?id=534587
and 
https://github.com/quarkusio/quarkus/issues/21226
(might be completely unrelated though)
However, the error message is not very clear for me.
Others suggest it is a classpath issue but how to fix?

**Comment by laeubi** [comment]
Any chance to reproduce the problem with a smaller example?

In general yes it is a classpath problem, but the reason is really hard to tell with such a large setup.

**Reply by carstenartur** [reply]
Not sure, does this smaller version help? It still contains several projects
https://github.com/carstenartur/sandbox/tree/reproducerclasspathissue

**Reply by carstenartur** [reply]
Ist it possible to switch on classpath Information for debugging pupose somehow?

**Reply by laeubi** [reply]
-X prints a  lot of information in this regard.

**Reply by carstenartur** [reply]
> -X prints a  lot of information in this regard.

Unless I missed something it does not give me any detailed information about the classpath or conflicts. I tried it already. Of course it is possible I made a mistake interpreting the log.

**Reply by carstenartur** [reply]
If the simplified project with only a single plugin and feature is still too much for an investigation - is there a working example somewhere to copy from? It just needs to be a use case building an eclipse derived ide using feature based configuration. And of course it should be a single plugin and feature in the best case.. :)

**Reply by laeubi** [reply]
> If the simplified project with only a single plugin and feature is still too much for an investigation

I think most people are on vacations, also it seems your problem is not "common" enough so other facing the same issue and can give some hints. Beside that, if it is crucial to your business you can find dedicated support options for Tycho here:
https://github.com/eclipse-tycho/tycho#getting-support

> is there a working example somewhere to copy from? It just needs to be a use case building an eclipse derived ide using feature based configuration. And of course it should be a single plugin and feature in the best case.. :)

Usually the integration test projects are a good source: https://github.com/eclipse-tycho/tycho/tree/master/tycho-its/projects you can checkout these and just search for your mojo of interest and will find some examples we use to test Tycho itself. Because of this, it is often the best way forward to provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue, that way we can make sure a fix is actually solving the problem and the use-case do not break in the future.
